reflect_extension.go

```diff
@@ -341,7 +341,7 @@ func describeStruct(ctx *ctx, typ reflect2.Type) *StructDescriptor {
 		if ctx.onlyTaggedField && !hastag && !field.Anonymous() {
 			continue
 		}
-		if tag == "-" {
+		if tag == "-" || field.Name() == "_" {
```

Comment:
If needed I can add another flag to `ConfigCompatibleWithStandardLibrary` to prevent changing the behaviour here for existing clients, which would also allow for them to still have access to these fields.